### PR TITLE
YAML test parser

### PR DESCRIPTION
**Pull request by tehampson**
YAML test parser that is agnostic as to what is actually running the parsed YAML tests

### CODE REVIEW

**Review by vivien-apple — APPROVED** [review]

File: scripts/tests/yamltests/fixes.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    as_bytes = [ord(c) for c in s]
+
+    if any([value > 0x200 for value in as_bytes]):
+        raise ValueError('Unsupported char in octet string %r' % as_bytes)
+    accumulated_hex = ''.join([f"{(v & 0xFF):02x}" for v in as_bytes])
+    return binascii.unhexlify(accumulated_hex)
+
+
+def try_add_yaml_support_for_scientific_notation_without_dot(loader):
+    regular_expression = re.compile(u'''^(?:
```
Comment (tehampson) [inline scripts/tests/yamltests/fixes.py:82]: @vivien-apple 

Can you please break this down a little more to something that looks like this
```
...
# Comment about what you are looking for here, and why
fixes_inifinity = '[-+]?\\.(?:inf|Inf|INF)'
# Comment about what you are looking for here, and why
fixes_nan = '\\.(?:nan|NaN|NAN)'

'|'.join([..., fixes_inifinity, fixes_nan])
```

File: scripts/tests/yamltests/parser.py [hunk 2]
```diff
@@ -0,0 +1,716 @@
...
+            return test_step
+
+        raise StopIteration
+
+
+class TestParser:
+    name: None
+    PICS: None
+    tests: None
+    _parsing_config_variable_storage: dict = {}
```
Comment (tcarmelveilleux) [inline scripts/tests/yamltests/parser.py:696]: This is not a standard way of expressing fields. Please have `__init__`  method declare the instance variable. These here are class variables and thus are global across instances of that class. Putting them at the top, and then overriding them in `__init__` is not needed and confusing. Please do this for all classes

File: scripts/tests/yamltests/parser.py [hunk 3]
```diff
@@ -0,0 +1,716 @@
...
+            return value
+
+
+class YamlTests:
+    _tests: list[_TestStepWithPlaceholders]
+    _index: 0
+
+    count: 0
+
+    def __init__(self, parsing_config_variable_storage: dict, definitions, tests: dict):
```
Comment (tcarmelveilleux) [inline scripts/tests/yamltests/parser.py:663]: Please document the API. None of the methods have docstrings

File: scripts/tests/yamltests/parser.py [hunk 4]
```diff
@@ -0,0 +1,716 @@
...
+
+class PostProcessCheckStatus(Enum):
+    '''Inidcates the post processing check step status.'''
+    SUCCESS = 'success',
+    WARNING = 'warning',
+    ERROR = 'error'
+
+
+class PostProcessCheckType(Enum):
+    '''Inidcates the post processing check step type.'''
```
Comment (tcarmelveilleux) [inline scripts/tests/yamltests/parser.py:89]: Typo in `Indicates` here and previous enum

**Review by tcarmelveilleux — APPROVED** [review]

File: scripts/tests/yamltests/fixes.py (hunk 1 above)
Comment (tehampson) [reply scripts/tests/yamltests/fixes.py:82]: Will be resolved in a follow up PR

**Pull request merged**